src/pybreaker.py

```diff
@@ -106,21 +106,32 @@ def reset_timeout(self, timeout):
         """
         self._reset_timeout = timeout
 
+    def _create_new_state(self, state, notify=True):
+        """
+        Return state object from state string, i.e.,
+        'closed' -> <CircuitClosedState>
+        """
+        state_map = {
+            STATE_CLOSED: CircuitClosedState,
+            STATE_OPEN: CircuitOpenState,
+            STATE_HALF_OPEN: CircuitHalfOpenState,
+        }
+        try:
+            return state_map[state](self, state, notify=notify)
```

Comment:
I have a hunch that this may be _slightly_ less performant than the `if/else` flow, but haven't tested it.

Reply:
Yeah, it is ever so slightly less performant however I think this falls under a level of optimization that yields little results, they're on the same magnitude:

```py
In [1]: def test_a(state):
   ...:     if state == 'foo':
   ...:         return 1
   ...:     elif state == 'bar':
   ...:         return 2
   ...:     else:
   ...:         return 3
   ...:

In [2]: def test_b(state):
   ...:     state_map = {
   ...:         'foo': 1,
   ...:         'bar': 2,
   ...:         'baz': 3
   ...:     }
   ...:     return state_map[state]
   ...:

In [3]: %%timeit
   ...: test_a('baz')
   ...:
The slowest run took 7.95 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 360 ns per loop

In [4]: %%timeit
   ...: test_b('baz')
   ...:
The slowest run took 5.69 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 503 ns per loop
```

Reply:
This won't be called in a 1000000-times loop, so I believe the performance penalty will be negligible. +1 to the dict version.

Reply:
awesome